selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/core/client/KeycloakAdminConsoleClient.java

```diff
@@ -0,0 +1,216 @@
+import static org.eclipse.che.selenium.core.CheSeleniumSuiteModule.OPENSHIFT_INFRASTRUCTURE;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.inject.Singleton;
+import org.eclipse.che.selenium.core.provider.RemovableUserProvider;
+import org.eclipse.che.selenium.core.user.AdminTestUser;
+import org.eclipse.che.selenium.core.user.DefaultTestUser;
+import org.eclipse.che.selenium.core.user.TestUser;
+import org.eclipse.che.selenium.core.user.TestUserFactory;
+import org.eclipse.che.selenium.core.user.TestUserImpl;
+import org.eclipse.che.selenium.core.utils.DockerUtil;
+import org.eclipse.che.selenium.core.utils.process.ProcessAgent;
+import org.eclipse.che.selenium.core.utils.process.ProcessAgentException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is java-client of 'keycloak/bin/kcadm.sh' command line application of keycloak container
+ *
+ * @author Dmytro Nochevnov
+ */
+@Singleton
+public class KeycloakAdminConsoleClient {
+  private static final Logger LOG = LoggerFactory.getLogger(KeycloakAdminConsoleClient.class);
+  private static final Pattern EXTRACT_USER_ID_PATTERN =
+      Pattern.compile("^.*Created new user with id '(.*)'.*$", Pattern.DOTALL);
+
+  @Inject private AdminTestUser adminTestUser;
+
+  private final DockerUtil dockerUtil;
+  private final TestUserFactory testUserFactory;
+  private final ProcessAgent processAgent;
+  private final String keycloakContainerId;
+  private final String cheInfrastructure;
+
+  @Inject
+  public KeycloakAdminConsoleClient(
+      DockerUtil dockerUtil,
+      TestUserFactory testUserFactory,
+      ProcessAgent processAgent,
+      @Named("che.infrastructure") String cheInfrastructure)
+      throws ProcessAgentException {
+    this.dockerUtil = dockerUtil;
+    this.testUserFactory = testUserFactory;
+    this.processAgent = processAgent;
+    this.cheInfrastructure = cheInfrastructure;
+
+    // obtain id of keycloak docker container
+    switch (cheInfrastructure) {
+      case OPENSHIFT_INFRASTRUCTURE:
+        this.keycloakContainerId =
+            processAgent.execute("echo $(docker ps | grep 'keycloak_keycloak-' | cut -d ' ' -f1)");
+        break;
+
+      case DOCKER_INFRASTRUCTURE:
+      default:
+        this.keycloakContainerId =
+            processAgent.execute("echo $(docker ps | grep che_keycloak | cut -d ' ' -f1)");
+        break;
+    }
+  }
+
+  public TestUserImpl createUser(RemovableUserProvider testUserProvider) throws IOException {
+    if (!dockerUtil.isCheRunLocally()) {
+      throw new IOException(
+          "It's impossible to create test user because of Che is running on the different host.");
+    }
+
+    long currentTimeInMillisec = System.currentTimeMillis();
+    String username = "user" + currentTimeInMillisec;
```

Comment:
What about using "NameGenerator" ?

Reply:
NameGenerator would be useful, but here stamp is used for the case when we will need to know test user time creation, for example, if some test user wasn't removed in time.